doc/user/content/sql/create-source/mysql.md

```diff
@@ -0,0 +1,387 @@
+
+
+If you're running MySQL as a managed service, further configuration changes
+might be required. For step-by-step instructions on enabling GTID-based binlog
+replication for your MySQL service, see the integration guides.
+
+#### Binlog retention
+
+{{< warning >}}
+If Materialize tries to resume replication and finds GTID gaps due to missing
+binlog files, the source enters an errored state and you have to drop and
+recreate it.
+{{< /warning >}}
+
+By default, MySQL retains binlog files for **30 days** (i.e., 2592000 seconds)
+before automatically removing them. This is configurable via the
+[`binlog_expire_logs_seconds`](https://dev.mysql.com/doc/mysql-replication-excerpt/8.0/en/replication-options-binary-log.html#sysvar_binlog_expire_logs_seconds)
+system variable. We recommend using the default value for this configuration in
+order to not compromise Materialize's ability to resume replication in case of
+failures or restarts.
+
+In some MySQL managed services, binlog expiration can be overriden by a
+service-specific configuration parameter. It's important that you double-check
+if such a configuration exists, and ensure it's set to the maximum interval
+available.
+
+As an example, [Amazon RDS for MySQL](/ingest-data/mysql-amazon-rds/) has its
+own configuration parameter for binlog retention ([`binlog retention hours`](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/mysql-stored-proc-configuring.html#mysql_rds_set_configuration-usage-notes.binlog-retention-hours))
+that overrides `binlog_expire_logs_seconds` and is set to `0` by default.
+
+#### Creating a source
+
+Materialize ingests the raw replication stream data for all (or a specific set
+of) tables in your upstream MySQL database.
+
+```sql
+CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_connection
+  FOR ALL TABLES;
+```
+
+When you define a source, Materialize will automatically:
+
+1. Create a **subsource** for each original table upstream, and perform an
+   initial, snapshot-based sync of the tables before it starts ingesting change
+   events.
+
+    ```sql
+    SHOW SOURCES;
+    ```
+
+    ```nofmt
+             name         |   type    |  size   |  cluster  |
+    ----------------------+-----------+---------+------------
+     mz_source            | mysql     |         |
+     mz_source_progress   | progress  |         |
+     table_1              | subsource |         |
+     table_2              | subsource |         |
+    ```
+
+1. Incrementally update any materialized or indexed views that depend on the
+   source as change events stream in, as a result of `INSERT`, `UPDATE` and
+   `DELETE` operations in the upstream MySQL database.
+
+It's important to note that the schema metadata is captured when the source is
+initially created, and is validated against the upstream schema upon restart.
+If you create new tables upstream after creating a MySQL source and want to
+replicate them to Materialize, the source must be dropped and recreated.
```

Comment:
should we say that this will be fixed by public preview?

Reply:
Added a note:

> {{< note >}}
> Support for dropping and recreating individual subsources is planned for a future
> release {{% gh 24975 %}}.
> {{< /note >}}